src/odemis/dataio/tiff.py

```diff
@@ -2722,3 +2744,168 @@ def _convertTFStoOdemisMD(metadata: str) -> dict:
+    # the SEM/FIB section header allows us to determine the beam type, 
+    # but it is corrupt and not parsed correctly by configparser. 
+    # Therefore we need to parse the metadata manually 
+    # and can't determine the beam type
+    # Once fixed:
+    # 1. Change to configparser to parse the metadata correctly
+    # 2. Correctly read the beam type from section headers
+    # 3. Remove the beam type check below (acq_type)
+
+    # parse ini metadata as dictionary 
+    # NOTE: not correctly ini formatted due to corruption
+    try:
+       tmd = _parse_tescan_metadata(metadata)
+    except Exception as e:
+        raise Exception(f"Failed to parse TESCAN metadata: {e}")
+
+    md = {}
+    
+    # tmp: inferred from metadata
+    beam_type = None
+
+    # general
+    try:
+        datetime_str = tmd["date"] + " " + tmd["time"]
+        date = datetime.strptime(datetime_str, "%Y-%m-%d %H:%M:%S")
+        md[model.MD_ACQ_DATE] = date.timestamp()
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_ACQ_DATE} - {e}")
+
+    try:
+        md[model.MD_HW_NAME] = tmd["device"] + "-" + tmd["serialnumber"]    
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_HW_NAME} -  {e}")
+    try:
+        md[model.MD_HW_VERSION] = tmd["device"] + "-" + tmd["devicemodel"]    
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_HW_NAME} -  {e}")
+    
+    try:
+        md[model.MD_SW_VERSION] = tmd["softwareversion"]
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_SW_VERSION} - {e}")
+
+    try:
+        md[model.MD_PIXEL_SIZE] = (float(tmd["pixelsizex"]), 
+                                   float(tmd["pixelsizey"]))
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_PIXEL_SIZE} - {e}")
+
+    try:
+        md[model.MD_DWELL_TIME] = float(tmd["dwelltime"])
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_DWELL_TIME} - {e}")
+    try:
+        # MD_POS should match image dimensions, e.g. only XY for 2D, XYZ for 3D, etc.
+        md[model.MD_POS] = (float(tmd["stagex"]), 
+                           float(tmd["stagey"]),
+                           )
+    except Exception as e:
+        logging.warning(f"Failed to parse TESCAN metadata: {model.MD_POS} - {e}")
+
+    # beam metadata
+    try:
+        # NOTE: due to metadata corruption, we can't directly read the beam type
+        # we check for the columntilt key which is only present in FIB metadata
+        # Once we can read the metadata correctly, we can remove this
+        # and directly read the beam type        
+        # acq_map =  {
+        #     "SEM": model.MD_AT_EM,
+        #     "FIB": model.MD_AT_FIB,
+        # }
+        # md[model.MD_ACQ_TYPE] = acq_map[beam_type]
```

Comment:
Is that correct? I couldn't find this info in the example metadata.

Reply:
Yeah, the columntilt key is only in FIB images, so it doesnt appear in SEM images. The example image is SEM